run.sh

```diff
@@ -1,33 +1,50 @@
 #!/bin/bash
 set -e
-cd `dirname $0`
-VENV_NAME=viam-venv
+
+UNAME=$(uname -s)
+
+VENV_NAME=".venv"
 echo "Current Directory: $(pwd)"
 echo "Checking for virtual environment folder..."
 
-if [ -d "viam-env" ]
-  then
-    echo "Virtual environment found, activating..."
-    source viam-env/bin/activate
-    echo "Virtual environment activated: $VENV_NAME"
-  else
-    echo "Setting up virtual environment..."
-    python3 -m venv --system-site-packages "$VENV_NAME"
-    source "$VENV_NAME/bin/activate"
-    echo "Virtual environment activated: $VENV_NAME"
-    echo "Installing dependencies from requirements.txt..."
-
-    while IFS= read -r requirement; do
-      if [ -n "$requirement" ]; then
-        echo "Installing $requirement..."
-        pip install "$requirement" || echo "Failed to install $requirement"
-      fi
-    done < requirements.txt
+if [ -d "$VENV_NAME" ]; then
+  echo "Virtual environment found, activating..."
+  source "$VENV_NAME/bin/activate"
 
-    echo "Dependencies installation complete."
+  # Check if 'uv' can be imported in Python
+  if ! python3 -c "import uv" > /dev/null 2>&1; then
+    echo "'uv' not found in virtual environment. Recreating the virtual environment..."
+    deactivate
+    rm -rf "$VENV_NAME"
+  else
+    echo "'uv' is installed, skipping recreation of virtual environment."
+  fi
+fi
 
+# Create a new virtual environment if it doesn't exist or was removed
+if [ ! -d "$VENV_NAME" ]; then
+  echo "Setting up virtual environment..."
+  
+  if [ "$UNAME" = "Linux" ]; then
+    echo "Installing uv on Linux"
+    pip install uv
```

Comment:
Should we be defensive and check if pip is installed on the linux runner?

Reply:
Actually I'm pretty sure it comes default on runners, but in the chance someone wants to build on RPI4 or something (unlikely) might be good to add. I'd say addressing the above comment is optional!

Reply:
I will add for thoroughness